doc/changelog/10-coqide/19417-coqide-preferences.rst

```diff
@@ -0,0 +1,22 @@
+- **Changed:**
+  Improved Preferences dialog: larger margins,
+  tree of categories, sections in the categories,
+  spin buttons for numbers, preservation of the last
+  selected category, and more
+  (`#19417 <https://github.com/coq/coq/pull/19417>`_,
+  by Sylvain Chiron).
+- **Fixed:**
+  All preferences are now applied after clicking Apply or OK
+  rather than immediately
+  (`#19417 <https://github.com/coq/coq/pull/19417>`_,
+  by Sylvain Chiron).
+- **Fixed:**
+  Changing the allowed modifiers in the Shortcuts panel of
+  the Preferences dialog now immediately updates the available
+  modifiers for the listed menus
+  (`#19417 <https://github.com/coq/coq/pull/19417>`_,
+  by Sylvain Chiron).
+- **Added:**
+  Preference setting for unjustified conclusions background color
+  (`#19417 <https://github.com/coq/coq/pull/19417>`_,
+  by Sylvain Chiron).
```

Comment:
> This looks odd. It didn't say that before. Previously it said `coq_style`. Would you change it to a better name?

It is in the `coq_style.xml` file. So let’s say “Coq style”?

> I'm also getting odd behavior when I click repeatedly on Style scheme. Frequently, when the dropdown isn't showing, the dropdown appears when I click and hold the mouse button and disappears when I release the button. Do you see the same behavior? It may be an artifact of my X-Windows, which is not very good.

I don’t get any strange behavior; the dropdown list appears all the time for me. If you hold the mouse button, then move the mouse and release the button, it is normal that the dropdown disappears immediately.

Reply:
I suppose good enough.  I think the Language setting is supposed to choose between `ide/coqide/coq.lang` and `.../coq-ssreflect`, which specifies how to do syntax highlighting.  I thought `using` would be highlighted as an SSR keyword, but I can't get it to happen. I didn't try very hard and these files look complex.  And I'm sure they're out of date (missing commands, tactics and keywords) not to mention that regular expressions can't do a perfect job highlighting keywords.

![image](https://github.com/user-attachments/assets/a15de655-39b4-4eec-82ac-ed6273e2214f)

Reply:
Yes, we already discussed last year that they are out of date, and it was also said that regular expressions cannot do a perfect job — though I still wonder why regexp aren’t enough, and PCREs allow a bit more than regexps, there are some look-ahead syntaxes allowed.

Reply:
Hmm, maybe too strong to say regexps can't do a perfect job.  I don't have a good example in hand.  Depends on what you want to highlight.

Adding lookahead to REs doesn't make them more powerful in theory.  There's always an equivalent non-lookahead RE.  But REs with lookahead may make them easier to work with.  REs are not able to correctly recognize only balanced parentheses, e.g. accepting "(())()" but not "(()".  To do that requires a context-free grammar and a parser stack.